zk/dnshostprovider.go

```diff
@@ -34,17 +34,19 @@ func (hp *DNSHostProvider) Init(servers []string) error {
 	for _, server := range servers {
 		host, port, err := net.SplitHostPort(server)
 		if err != nil {
-			return err
+			// If one node is down, Init should still continue
```

Comment:
why this change? in this case the node isn't down - instead it's a configuration error. shouldn't this fail-fast?

Reply:
Good call. If it's a configuration error, we should fail fast here